pkg/command/root.go

```diff
@@ -125,6 +131,10 @@ func newRootCmd() *cobra.Command {
 		// Flag parsing must be deactivated because the root plugin won't know about all flags.
 		DisableFlagParsing: true,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			if err := telemetry.Client().UpdateCmdPreRunMetrics(cmd, args); err != nil {
```

Comment:
Do we really need to make telemetry calls synchronous? How about making the telemetry calls` (UpdateCmdPreRunMetrics, UpdateCmdPostRunMetrics, SaveMetrics, and SendMetrics)` asynchronous using goroutines? Since the CLI command will continue irrespective of these telemetry calls, making them asynchronous can improve performance and responsiveness.
cc: @anujc25

Reply:
`UpdateCMDPostRunMetrics`,SaveMetrics` and SendMetrics` are sequential and should be called after the CLI command(core/plugin) is executed, as we need to collect the metrics after the CLI command is execited. `UpdateCmdPreRunMetrics` is also not blocking/time consuming(since we already have command tree cache). So, I don't see a necessity for asynchronous calls at the moment.